src/lib/types.ts

```diff
@@ -209,14 +210,335 @@ type ResolveInviteCodeResponse = {
+  endDate: Date; // allocated time of a space
+  publicStartDate?: Date; // public facing
+  publicEndDate?: Date; // public facing
+  resources: ResourceRequestId[];
+  links: Link[];
+  images: Image[];
+};
+
+/**
+ * User
+ *
+ * NOTE: sent on `inbox` channel
+ */
+
+type UserNameAssigned = {
+  type: "user_name_assigned";
   data: {
     name: string;
-    startDate?: string;
-    endDate?: string;
+  };
+};
+
+/**
+ * Access
+ *
+ * NOTE: sent on `inbox` channel
+ */
+
+type CalendarAccessRequested = {
+  type: "calendar_access_requested";
+  data: {
+    calendarId: Hash;
+    name: string;
+    message: string;
+  };
+};
+
+type CalendarAccessAccepted = {
+  type: "calendar_access_accepted";
+  data: {
+    calendarId: Hash;
+  };
+};
+
+type CalendarAccessRejected = {
+  type: "calendar_access_rejected";
+  data: {
+    calendarId: Hash;
+  };
+};
+
+/**
+ * Calendar
+ */
+
+type CalendarCreated = {
+  type: "calendar_created";
+  data: {
+    fields: CalendarFields;
+  };
+};
+
+type CalendarUpdated = {
+  type: "calendar_updated";
+  data: {
+    id: Hash;
+    fields: CalendarFields;
+  };
+};
+
+type SpacesPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type ResourcesPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type AboutPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type CalendarDeleted = {
+  type: "calendar_deleted";
+};
+
+/**
+ * Space
+ */
+
+type SpaceCreated = {
+  type: "space_created";
+  data: {
+    field: SpaceFields;
+  };
+};
+
+type SpaceUpdated = {
+  type: "space_updated";
+  data: {
+    id: Hash;
+    fields: SpaceFields;
+  };
+};
+
+type SpaceDeleted = {
+  type: "space_deleted";
+};
+
+/**
+ * Resource
+ */
+
+type ResourceCreated = {
+  type: "resource_created";
+  data: {
+    fields: ResourceFields;
+  };
+};
+
+type ResourceUpdated = {
+  type: "space_updated";
+  data: {
+    id: Hash;
+    fields: ResourceFields;
+  };
+};
+
+type ResourceDeleted = {
+  type: "resource_deleted";
+};
+
+/**
+ * Event
+ */
+
+type EventCreated = {
+  type: "event_created";
+  data: {
+    fields: EventFields;
+  };
+};
+
+type EventUpdated = {
+  type: "event_updated";
+  data: {
+    id: Hash;
+    fields: EventFields;
+  };
+};
+
+type EventDeleted = {
+  type: "event_deleted";
+  data: {
+    id: Hash;
+  };
+};
+
+/**
+ * Requests and responses
+ */
+
+type SpaceRequested = {
+  type: "space_requested";
+  data: {
+    eventId: Hash;
+    spaceId: Hash;
+    message: string;
+    timePeriod: TimeSpan[];
+  };
+};
+
+type SpaceRequestAccepted = {
+  type: "space_request_accepted";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type SpaceRequestRejected = {
+  type: "space_request_rejected";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type SpaceRequestCancelled = {
```

Comment:
This message type is for when a resource was requested, but then either the user cancels that request manually, or deletes the event which a resource was attached to.

Reply:
Similar to the other point above I'd only use this event for cancelling a request manually (detecting if an event was deleted needs to be done by the requester)